pkg/apis/core/v1/defaults.go

```diff
@@ -142,6 +142,19 @@ func SetDefaults_Service(obj *v1.Service) {
 			obj.Spec.AllocateLoadBalancerNodePorts = pointer.Bool(true)
 		}
 	}
+
+	if obj.Spec.Type == v1.ServiceTypeLoadBalancer {
+		if utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) {
+			ipMode := v1.LoadBalancerIPModeVIP
+
+			for i, ing := range obj.Status.LoadBalancer.Ingress {
+				if ing.IP != "" && ing.IPMode == nil {
+					obj.Status.LoadBalancer.Ingress[i].IPMode = &ipMode
+				}
+			}
+		}
+	}
```

Comment:
I don't think we should default, just validate that if is set is a valid value only

Reply:
Why not?  If we don't set a default, then every consumer needs to hand-code an "else" path which will be functionally identical to "VIP" ?

Reply:
... and what happen with existing code? all existing code will have to adapt, no?

Reply:
and this is only set if already ` obj.Status.LoadBalancer.Ingress `. is set, I think I get it now